### PR TITLE
HPCC-17811 configmgr terminates unexpectedly

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@afishbeck Please can you review this.

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 1]
```diff
@@ -2914,7 +2914,9 @@ bool CWsDeployFileInfo::displaySettings(IEspContext &context, IEspDisplaySetting
       IPropertyTree* pFolder = &iter->query();
 
       const char* pszCompType = pFolder->queryProp("@compType");
+      if (!pszCompType) continue;
```
Comment (afishbeck) [inline esp/services/WsDeploy/WsDeployService.cpp]: quite trivial but we'd usually put "continue;" on the next line.

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 2]
```diff
@@ -2914,7 +2914,9 @@ bool CWsDeployFileInfo::displaySettings(IEspContext &context, IEspDisplaySetting
...
       const char* pszCompName = pFolder->queryProp(XML_ATTR_NAME);
+      if (!pszCompName) continue;
```
Comment (afishbeck) [inline esp/services/WsDeploy/WsDeployService.cpp]: see above.

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 3]
```diff
@@ -2949,6 +2951,7 @@ bool CWsDeployFileInfo::displaySettings(IEspContext &context, IEspDisplaySetting
 
       const char* buildSetName = pBuildSet->queryProp(XML_ATTR_NAME);
       const char* processName = pBuildSet->queryProp(XML_ATTR_PROCESS_NAME);
+      if (!buildSetName || !processName) continue;
```
Comment (afishbeck) [inline esp/services/WsDeploy/WsDeployService.cpp]: see above.

**Review by afishbeck — COMMENTED** [review]
@shamser other than trivial formatting looks good.

**Comment by shamser** [comment]
@afishbeck Reformatted as per comment. Everything else ok?

**Comment by afishbeck** [comment]
@shamser yes, looks fine

**Pull request merged**